src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -102,10 +103,20 @@ public virtual Task<bool> ShouldHandleRequestAsync()
                         throw new Exception("An error was returned from the RemoteFailure event.", errorContext.Result.Failure);
                     }
                 }
-
+                properties = errorContext.Properties ?? new AuthenticationProperties();
                 if (errorContext.Failure != null)
                 {
-                    throw new Exception("An error was encountered while handling the remote login.", errorContext.Failure);
+                    var builder = new QueryBuilder();
+                    if (properties.Items.TryGetValue(".error", out var error))
+                    {
+                        builder.Add("error", error);
+                    }
+                    if (properties.Items.TryGetValue(".error_description", out var errorDescription))
+                    {
+                        builder.Add("error_description", errorDescription);
+                    }
+                    Context.Response.Redirect(Options.RemoteFailureRedirect + builder.ToQueryString());
```

Comment:
Should there be a way to disable this redirect?  Or what if there is no redirect url?

Reply:
You can handle the event above to avoid the redirect.

Reply:
Fair enough, we still need to validate or ignore a null RemoteFailureRedirect though